src/mpi/romio/adio/ad_daos/ad_daos_open.c

```diff
@@ -45,8 +45,14 @@ static int parse_filename(const char *path, char **_obj_name, char **_cont_name)
                 rc = ENOMEM;
                 goto out;
             }
+            *_cont_name = cont_name;
+        } else {
+            *_cont_name = ADIOI_Strdup(cont_name);
+            if (*_cont_name == NULL) {
+                rc = ENOMEM;
+                goto out;
+            }
```

Comment:
You just need to change the condition at line 32 into `if (strcmp(cont_name, ".") == 0)`, right?

Reply:
yes you are right.
but this case requires some more handling under this if condition to support properly which we don't yet, which is why i had it that way to just avoid the double free crash for now. anyway i will update this now.